docs/edgeql/group.rst

```diff
@@ -0,0 +1,195 @@
+.. _ref_eql_group:
+
+Group
+=====
+
+EdgeQL supports a top-level ``group`` statement. These are used to partition
+sets into subsets based on some parameters. These subsets then can be
+additionally aggregated to provide some analytics.
+
+The most basic format is just using the bare :eql:stmt:`group` to group a set
+of objects by some property:
+
+.. code-block:: edgeql-repl
+
+    db> with M := (select Movie filter .release_year > 2015)
+    ... group M{title} by .release_year;
```

Comment:
In the rest of the docs I've been adding a space before the left curly brace (`M {title}`) even if it's "inline" like this. There are a handful of other occurrences of `M{title}` in the reference docs